INV4/pallet-ips/src/tests.rs

```diff
@@ -200,12 +208,414 @@ fn destroy_should_fail() {
+            IpsStorage::<Runtime>::get(1),
+            Some(IpsInfoOf::<Runtime> {
+                parentage: Parentage::Parent(multi_account_id::<Runtime, IpsId>(1, None)),
+                metadata: MOCK_METADATA.to_vec().try_into().unwrap(),
+                data: vec![AnyId::IpfId(0)].try_into().unwrap(),
+                ips_type: IpsType::Replica(0),
+                allow_replica: false,
+            })
+        );
+
+        assert_noop!(
+            Ips::create_replica(Origin::signed(BOB), 1),
+            Error::<Runtime>::ReplicaNotAllowed
+        );
+
+        assert_eq!(Ips::next_ips_id(), 2);
+
+        // Case 4: Original Ips does not exist
+        assert_noop!(
+            Ips::create_replica(Origin::signed(BOB), 2),
+            Error::<Runtime>::IpsNotFound
+        );
+
+        // Case 5: IpsId Overflow
+        NextIpsId::<Runtime>::mutate(|id| *id = IpsId::max_value());
+        assert_noop!(
+            Ips::create_replica(Origin::signed(BOB), 0),
+            Error::<Runtime>::NoAvailableIpsId
+        );
+    });
+}
+
+#[test]
+fn allow_replica_should_work() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(Ipf::mint(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            H256::from(MOCK_DATA)
+        ));
+
+        assert_eq!(Ips::next_ips_id(), 0);
+        assert_ok!(Ips::create_ips(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            vec![0],
+            false,
+        ));
+
+        assert_ok!(Ips::allow_replica(
+            Origin::signed(multi_account_id::<Runtime, IpsId>(0, None)),
+            0
+        ));
+
+        assert_eq!(
+            IpsStorage::<Runtime>::get(0),
+            Some(IpsInfoOf::<Runtime> {
+                allow_replica: true,
+                parentage: Parentage::Parent(multi_account_id::<Runtime, IpsId>(0, None)),
+                metadata: MOCK_METADATA.to_vec().try_into().unwrap(),
+                data: vec![AnyId::IpfId(0)].try_into().unwrap(),
+                ips_type: IpsType::Normal
+            })
+        )
+    })
+}
+
+#[test]
+fn allow_replica_should_fail() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(Ipf::mint(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            H256::from(MOCK_DATA)
+        ));
+
+        assert_eq!(Ips::next_ips_id(), 0);
+        assert_ok!(Ips::create_ips(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            vec![0],
+            false,
+        ));
+
+        // Case 0: Extrinsic called in a non-multisig context:
+        assert_noop!(
+            Ips::allow_replica(Origin::signed(ALICE), 0),
+            Error::<Runtime>::NoPermission
+        );
+
+        // Case 1: An unknown origin tries to allow replica on IPS 0:
+        assert_noop!(
+            Ips::allow_replica(Origin::none(), 0),
+            DispatchError::BadOrigin,
+        );
+
```

Comment:
MIssing the following cases:
- IPS doesn't exist
- IPS is not a Parent
- IPS already allows replica
- IPS is a replica and therefore cannot be replicated